File: accounting_app_ui/lib/models/auth_models.dart
class LoginResponse {
  final String token;
  final String username;
  final DateTime expiration;
  final List<String> roles;
  final List<String> permissions;

  LoginResponse({
    required this.token,
    required this.username,
    required this.expiration,
    required this.roles,
    required this.permissions,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'],
      username: json['username'],
      expiration: DateTime.parse(json['expiration']),
      roles: List<String>.from(json['roles']),
      permissions: List<String>.from(json['permissions'] ?? json['Permissions'] ?? []),
    );
  }
}

File: AccountingApp.API/Models/AuthModels.cs
using System.ComponentModel.DataAnnotations;

namespace AccountingApp.API.Models
{
    public class LoginRequest
    {
        [Required]
        public string Username { get; set; } = string.Empty;

        [Required]
        public string Password { get; set; } = string.Empty;
    }

    public class LoginResponse
    {
        public string Token { get; set; } = string.Empty;
        public DateTime Expiration { get; set; }
        public string Username { get; set; } = string.Empty;
        public List<string> Roles { get; set; } = new List<string>();
    }
}
